common/flatpak-dir.c

```diff
@@ -12438,66 +12434,91 @@ find_matching_refs (GHashTable           *refs,
   return g_steal_pointer (&matched_refs);
 }
 
+static GPtrArray *
+get_refs_for_arch (GPtrArray *refs, const char *arch)
+{
+  g_autoptr(GPtrArray) arched_refs = g_ptr_array_new_with_free_func ((GDestroyNotify)flatpak_decomposed_unref);
+
+  for (int i = 0; i < refs->len; i++)
+    {
+      FlatpakDecomposed *ref = g_ptr_array_index (refs, i);
+      if (flatpak_decomposed_is_arch (ref, arch))
+        g_ptr_array_add (arched_refs, flatpak_decomposed_ref (ref));
+    }
+
+  return g_steal_pointer (&arched_refs);
+}
+
+static gpointer
+fail_multiple_refs (GError    **error,
+                    const char *name,
+                    GPtrArray  *refs)
+{
+  g_autoptr(GString) err = g_string_new ("");
+
+  g_string_printf (err, _("Multiple branches available for %s, you must specify one of: "), name);
```

Comment:
This error message isn't quite right in case the branches are the same but the arches are different?

Reply:
Yeah, i noticed this because i left the old message. However, i think in practice that the error describes the common situation when this error is hit. I'm not sure complicating it for some edge case helps. We still list the actual refs so you can see what goes wrong.